modules/examples/sources/msg-generator/msg-generator-source.c

```diff
@@ -93,13 +93,27 @@ _free(LogPipe *s)
   log_source_free(s);
 }
 
+static void
+_add_name_value(gpointer key, gpointer value, gpointer data)
+{
+  gchar *name = (gchar *) key;
+  LogTemplate *val = (LogTemplate *) value;
+  LogMessage *msg = (LogMessage *) data;
+  GString *msg_body = g_string_sized_new(128);
```

Comment:
It would be good to use `ScratchMarkers` here.
Example: https://github.com/syslog-ng/syslog-ng/blob/master/modules/dbparser/correllation-context.c#L53
Key points are `Marker`, `alloc_and_mark()`, `reclaim_marked()`. Simple `alloc()` is not needed here, as we only need one `GString *`.

Or, even better, get only one `GString *` with `ScratchBuffers` in `send_generated_message()` and pass it to `_add_name_value()`, which will truncate it to 0 size before it uses it.

Each solution is okay for me, but `g_string_new()` is avoidable, if possible, because of performance reasons.

Reply:
We were thinking about this together with @alltilla , and concluded that the original implementation is fine. The mark-reclaim api might work, but the simple scratch_buffers_alloc might not, because only `log_msg_post` and a few places are protected by garbage collection. The performance benefit does not worth the complexity. Marking as resolved.

Reply:
Although it is true that scratch buffers can save a lot of cycles for hot paths, I don't think this is one of those. msg-generator is at best a testing tool and I doubt it would generate a huge rate of messages per second and scratch buffers only play a role in those scenarios.

Reply:
Sure, I agree, but as it is an "example" driver, it could show us an example, and it would be nice to see the building blocks of our real drivers. I might have the wrong idea about the examples folder.

I still agree with furiel though, as if we need to dig deep to make sure it works in this case, it is not worth the time and complexity.